x-pack/plugins/observability/server/assistant_functions/get_slo_change_point.ts

```diff
@@ -0,0 +1,297 @@
+      type: 'string',
+      description: 'Optional start of an alert associated with the given SLO',
+    },
+    metricOperation: {
+      type: 'string',
+      description: `Optional operation to apply to the metric field for analysis. Defaults to 'avg' for average.
+        * Average: 'avg'
+        * Maximum: 'max'
+        * Minimum: 'min
+        * Sum: 'sum`,
+      items: {
+        type: 'string',
+        enum: ['avg', 'max', 'min', 'sum'],
+      },
+    },
+  },
+  required: ['slo.name'],
+} as const;
+
+export function registerGetSLOChangeDetectionFunction({
+  sloClient,
+  resources,
+  registerFunction,
+  dataViewsClient,
+}: FunctionRegistrationParameters) {
+  registerFunction(
+    {
+      name: 'get_slo_change_point',
+      contexts: ['slo'],
+      description: `
+        Gets the source index and detects changes in the source data for an SLO. 
+        
+        Visualise changes in the underlying data for a given SLO. A visualisation will be displayed above your reply - DO NOT attempt to display or generate an image yourself, or any other placeholder. Additionally, the function will return any changes, such as spikes, step and trend changes, or dips.
+        
+        Include the observed period. For example, if the observed period is "24h" the output should be "over the past 24 hours".
+
+        Always include the 'metricOperation' as well, in plain english, before the field names. For example, if the 'metricOperation' is "avg", this is the "average" and applies to all fields.
+        Example "average field1, field2, and field3"
+
+        After presenting the user with the change detection data, add a new line and be sure to ask if they want to repeat the analysis
+        over a different time period, metric operation, or SLO.
+      `,
+      descriptionForUser: i18n.translate(
+        'xpack.observability.observabilityAiAssistant.functions.registerGetSLOList.descriptionForUser',
+        {
+          defaultMessage: `Gets the source index and changes in the source data for an SLO.
+          `,
+        }
+      ),
+      parameters,
+    },
+    async ({ arguments: args }, signal) => {
+      const {
+        'slo.name': name,
+        shortWindow,
+        longWindow,
+        field,
+        start,
+        end,
+        'kibana.alert.start': alertStart,
+        metricOperation = 'avg',
+      } = args;
+      console.log('aggregationType', metricOperation);
+      const params = {
+        kqlQuery: `slo.name: "${name}*"`,
+        sortBy: 'status',
+        sortDirection: 'desc',
+        page: '1',
+        perPage: '25',
+      };
+      console.log('start', start);
+      console.log('end', end);
+      console.log('shortWindow', shortWindow);
+      console.log('longWindow', longWindow);
+      console.log('alertStart', alertStart);
+      const slos = await sloClient.find.execute(params);
+      let absoluteStartTime: string | undefined;
+
+      if (alertStart && longWindow) {
```

Comment:
The function also optionally takes the start time and long burn rate window if there's an associated alert. If these options are passed, it attempts to create a time range to run analysis based on the alert. Otherwise, it uses `now-24h` by default. The LLM can also specify a time range.